src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -63,19 +63,27 @@ class ProductImageRepository extends AbstractObjectModelRepository
      */
     private $productImageValidator;
 
+    /**
+     * @var int
+     */
+    private $contextShopId;
+
     /**
      * @param Connection $connection
      * @param string $dbPrefix
      * @param ProductImageValidator $productImageValidator
+     * @param int $contextShopId
      */
     public function __construct(
         Connection $connection,
         string $dbPrefix,
-        ProductImageValidator $productImageValidator
+        ProductImageValidator $productImageValidator,
+        int $contextShopId
```

Comment:
I don't think repository should be context dependant This is the kind of nightmare that makes all the legacy code so hard to maintain and debug If we handle multishop it must be via an explicitly provided parameter

But we didn't plan on handling multi shop in product page for now so you can remove this parameter along with the SQL condition on `image_shop`

Reply:
okay ill remove this and add todo "multishop not handled"